Jenkinsfile

```diff
@@ -76,13 +76,31 @@ def cleanup_master() {
 	}
 }
 
+def archive_logs() {
+	sh '''
+	cd "$(echo "$WORKSPACE" | cut -f 1 -d '@')"
+	# need a unique and descriptive name so that the artifacts don't clobber each other
+	# and we can know which log file goes to which node, etc
+	# this also removes a / that would otherwise be included
+	LOGS_NAME=$(echo "logs-$JOB_NAME-$NODE_NAME-$BUILD_ID" | sed 's/\\///g')
+	mv logs "$LOGS_NAME"
+
+	if [ "$(pwd)" != "$WORKSPACE" ]; then
+		rm -rf "$WORKSPACE"/logs* || true
+		cp -r "$LOGS_NAME" "$WORKSPACE"
+	fi
+	'''
+	archiveArtifacts artifacts: "logs-*/*"
+}
+
 def run_action(action) {
 	try {
 		sh """
 		cd "\$(echo ${env.WORKSPACE} | cut -f 1 -d '@')"
 		npm run ${action}
 		"""
 	} catch (err) {
+		archive_logs()
```

Comment:
It looks like we're calling `archive_logs` both here in the `catch` block of `run_action` and after calls to `run_action` (e.g. line 290): if it needs to be called after `run_action` no matter the outcome then let's put it a `finally` block.